### DISCUSSION TITLE
batch size 1 throws index error

### DISCUSSION THREAD

**Discussion by alex-de-zwart** [pyg-team/pytorch_geometric#4758] [General]
I'd like to benchmark different models also in terms of latency, but when I use a batch size of 1 it torch geometric message passing throws the following error:

Dimension out of range (expected to be in range of [-1, 0], but got -2)
Edge_index is created with the following code

class GraphDataset(pyg_data.InMemoryDataset):
    def __init__(self, root, file_name, transform=None, pre_transform=None):
        self.file_name = file_name
        super().__init__(root, transform, pre_transform)
        self.data, self.slices = torch.load(self.processed_paths[0])

    @property
    def raw_file_names(self):
        return [f'{self.file_name}.txt']

    @property
    def processed_file_names(self):
        return [f'{self.file_name}.pt']

    def download(self):
        pass

    def process(self):
        raw_data_file = f'{self.raw_dir}/{self.raw_file_names[0]}'
        print(raw_data_file)
        with open(raw_data_file, 'rb') as f:
            sessions = pickle.load(f)
        data_list = []

        for session in range(len(sessions[0])):
          session, y = sessions[0][session], sessions[1][session]
          codes, uniques = pd.factorize(session)
          senders, receivers = codes[:-1], codes[1:]

          # Build Data instance
          edge_index = torch.tensor([senders, receivers], dtype=torch.long)
          x = torch.tensor(uniques, dtype=torch.long).unsqueeze(1)
          y = torch.tensor([y[0]], dtype=torch.long)
          data_list.append(pyg_data.Data(x=x, edge_index=edge_index, y=y))
        
        data, slices = self.collate(data_list)
        torch.save((data, slices), self.processed_paths[0])

**Comment by rusty1s** [comment]
This likely indicates that `x` is only one-dimensional (although I see that you explicitly `unsqueeze(1)` already). Can you still confirm that this is not the case?

**Reply by alex-de-zwart** [reply]
Hi @rusty1s thanks for the prompt reply much appreciated!

Before x is passed to a gru cell it's squeezed again, similar to what happens in this notebook:

https://colab.research.google.com/drive/1XwQ0njqSZL8vbHJMnRRHlH4ar0kYYFVz?usp=sharing#scrollTo=TNsAHq8PCN3k

**Reply by rusty1s** [reply]
Ok, what does that mean? Whenever you pass `x` to a `MessagePassing` module, it should be two-dimensional.
